hub/util/transform.py

```diff
@@ -133,16 +131,16 @@ def merge_tensor_metas(
         for all_tensor_meta in all_workers_tensor_meta:
             current_meta = all_tensor_meta[tensor]
             # will be None if 0 outputs from worker
-            if tensor_meta.dtype is None:
+            if tensor_meta.max_shape == [] or tensor_meta.dtype is None:
```

Comment:
Probably better to check `len(tensor_meta.max_shape)` in case we ever switch it to tuple or a custom object